src/DIRAC/WorkloadManagementSystem/Agent/StatesAccountingAgent.py

```diff
@@ -66,17 +75,52 @@ def initialize(self):
 
     def execute(self):
         """Main execution method"""
-        # Get the WMS Snapshot!
+
+        # PilotsHistory to Monitoring
+        if self.monitoringEnabled:
+            self.log.info("Committing PilotsHistory to Monitoring")
+            result = PilotAgentsDB.getSummarySnapshot(self.__pilotKeyFields)
+            now = Time.dateTime()
+            if not result["OK"]:
+                self.log.error(
+                    "Can't get the PilotAgentsDB summary",
+                    "%s: won't commit PilotsHistory at this cycle" % result["Message"],
+                )
+                return S_ERROR()
+
+            values = result["Value"][1]
+            for record in values:
+                record = record[1:]
+                rD = {}
+                for iP in enumerate(self.__pilotKeyFields):
```

Comment:
I don't think this is how `enumerate` should be used here. For clarification:

```
In [1]: for i in range(len([1,2,3])):
   ...:     print(i)
   ...: 
0
1
2

In [2]: for i in enumerate([1,2,3]):
   ...:     print(i)
   ...: 
(0, 1)
(1, 2)
(2, 3)

In [3]: for i, _ in enumerate([1,2,3]):
   ...:     print(i)
   ...: 
0
1
2
```

I see you have taken (copied) this snippet of code from the WMSHistory part, please verify also there.